[sil_tai_dam_lao] New keyboard

JBrase opened the pull request:
Designed for typing the Tai Dam language in the Lao script. This update fills out unassigned keys with Lao characters that are not normally used for Tai Dam, adds the Lao numerals, adds an on-screen keyboard, and updates the documentation.

DavidLRowe commented:
readm.htm should probably be renamed to readme.htm. (I think it works the way it is, but it might save confusion later if it were renamed.)

With that change, I think this keyboard is ready to go.

darcywong00 commented on release/sil/sil_tai_dam_lao/sil_tai_dam_lao.kpj:
```diff
@@ -0,0 +1,194 @@
+      <Details>
+        <Name>Tai Dam Lao (SIL)</Name>
+        <Copyright>© SIL International 2016-2018</Copyright>
+        <Message>Keybaord is optimized for typing the Tai Dam langugage with the Lao script</Message>
```
typo on "Keyboard"

DavidLRowe replied:
Good catch! Also in .kmn file. (If it's corrected there, will the project file pick up the change?)

darcywong00 commented:
I had questions about the Saysettha MX font?
Did it originally come from http://laoscript.net/download/saysettha_mx.ttf ?

1. I wonder if the lower-cased filename with underscore is preferable.
2. Are there any accompanying license files that can be included? (README, FONTLOG.txt, OFL.txt)

darcywong00 commented on release/sil/sil_tai_dam_lao/source/sil_tai_dam_lao.kvks:
```diff
@@ -0,0 +1,8 @@
+  <header>
+    <version>10.0</version>
+    <kbdname>sil_tai_dam_lao</kbdname>
+    <flags/>
```
This OSK looks blank. When I tried to "Fill from layout" in Keyman Developer, it seemed to hang on **Importing Unshifted K_SLASH**.

darcywong00 commented on release/sil/sil_tai_dam_lao/source/welcome/welcome.htm:
```diff
@@ -0,0 +1,679 @@
+
+<h2>Font</h2>
+
+<p>The Tai Dam Lao (SIL) keyboard can be used with any font that includes the characters from the Lao Unicode block. For your convenience, the Saysettha MX font is included in the keyboard package. Saysettha MX is copyright &#x00A9 2015 by John M Durdin, and licensed under the <a href="https://scripts.sil.org/OFL">SIL Open Font License, Version 1.1</a>.</p>
```
For html, `&#x00A9` needs to be changed to `&amp;#x00A9`.

JBrase commented:
On 7/20/2018 4:09 AM, Darcy Wong wrote:
    
    
      
      I had questions about the Saysettha MX font?
        Did it originally come from http://laoscript.net/download/saysettha_mx.ttf
        ?
    
    yes
    
    
      
        I wonder if the lower-cased filename with underscore is
          preferable.
      
    
    I left the file name unaltered from the way the publisher named it.
    I noticed there were other shared fonts that did the same thing
    (canadian_aboriginal) and I did not know if I was permitted to alter
    it. I can change it if you want me to.
    
    
      
        Are there any accompanying license files that can be
          included? (README, FONTLOG.txt, OFL.txt)
      
    
    Not if you download only Saysettha MX. One can download Saysettha as
    part of the Laoscript 8 package, but then you get other software
    installed as well as a number hacked fonts that I don't want to
    encourage.
    
    Again, I didn't know if I should add anything to another party's
    product.
    
    Jim

JBrase commented:
fixed
    
    On 7/20/2018 3:58 AM, Darcy Wong wrote:
    
    
      
      @darcywong00 commented on this pull request.
      
      In release/sil/sil_tai_dam_lao/sil_tai_dam_lao.kpj:
      > +  <Options>
+    <BuildPath>$PROJECTPATH\build</BuildPath>
+    <CompilerWarningsAsErrors>True</CompilerWarningsAsErrors>
+    <WarnDeprecatedCode>True</WarnDeprecatedCode>
+  </Options>
+  <Files>
+    <File>
+      <ID>id_3ae844e95f33afc5b249d76fc77f418e</ID>
+      <Filename>sil_tai_dam_lao.kmn</Filename>
+      <Filepath>source\sil_tai_dam_lao.kmn</Filepath>
+      <FileVersion>2.0</FileVersion>
+      <FileType>.kmn</FileType>
+      <Details>
+        <Name>Tai Dam Lao (SIL)</Name>
+        <Copyright>© SIL International 2016-2018</Copyright>
+        <Message>Keybaord is optimized for typing the Tai Dam langugage with the Lao script</Message>

      typo on "Keyboard"
      —
        You are receiving this because you authored the thread.
        Reply to this email directly, view it on GitHub, or mute the thread.
      {"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/keymanapp/keyboards","title":"keymanapp/keyboards","subtitle":"GitHub repository","main_image_url":"https://assets-cdn.github.com/images/email/message_cards/header.png","avatar_image_url":"https://assets-cdn.github.com/images/email/message_cards/avatar.png","action":{"name":"Open in GitHub","url":"https://github.com/keymanapp/keyboards"}},"updates":{"snippets":[{"icon":"PERSON","message":"@darcywong00 commented on #403"}],"action":{"name":"View Pull Request","url":"https://github.com/keymanapp/keyboards/pull/403#pullrequestreview-138978000"}}}
      [
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/keymanapp/keyboards/pull/403#pullrequestreview-138978000",
"url": "https://github.com/keymanapp/keyboards/pull/403#pullrequestreview-138978000",
"name": "View Pull Request"
},
"description": "View this Pull Request on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
},
{
"@type": "MessageCard",
"@context": "http://schema.org/extensions",
"hideOriginalBody": "false",
"originator": "AF6C5A86-E920-430C-9C59-A73278B5EFEB",
"title": "@darcywong00 commented on 403",
"sections": [
{
"text": "",
"activityTitle": "**Darcy Wong**",
"activityImage": "https://assets-cdn.github.com/images/email/message_cards/avatar.png",
"activitySubtitle": "@darcywong00",
"facts": [

]
}
],
"potentialAction": [
{
"targets": [
{
"os": "default",
"uri": "https://github.com/keymanapp/keyboards/pull/403#pullrequestreview-138978000"
}
],
"@type": "OpenUri",
"name": "View on GitHub"
},
{
"name": "Unsubscribe",
"@type": "HttpPOST",
"target": "https://api.github.com",
"body": "{\n\"commandName\": \"MuteNotification\",\n\"threadId\": 357718307\n}"
}
],
"themeColor": "26292E"
}
]

JBrase commented:
On 7/20/2018 4:23 AM, Darcy Wong wrote:
    
    
      
      @darcywong00 commented on this pull request.
      
      In release/sil/sil_tai_dam_lao/source/welcome/welcome.htm:
      > +							 
+						"Saysettha MX"; 
+          position: absolute; 
+          display: block;
+          right: 5px;
+          bottom: 4px;
+          color: blue;
+        }
+    </style>
+</head>
+<body>
+<h1>Tai Dam Lao (SIL) keyboard package</h1>
+
+<h2>Font</h2>
+
+<p>The Tai Dam Lao (SIL) keyboard can be used with any font that includes the characters from the Lao Unicode block. For your convenience, the Saysettha MX font is included in the keyboard package. Saysettha MX is copyright &#x00A9 2015 by John M Durdin, and licensed under the <a href="https://scripts.sil.org/OFL">SIL Open Font License, Version 1.1</a>.</p>

      For html, &#x00A9 needs to be changed to &amp;#x00A9.
    
    I don't think so. That will not render properly. U+00A9 is the
    copyright symbol.
    
    
    
    I guess I could use &copy.

DavidLRowe commented:
Fill from layout worked for me. There are a lot of blank key tops since the .kmn file doesn't include keys that aren't changed.

JBrase commented:
On 7/20/2018 4:17 AM, Darcy Wong wrote:
    
    
      
      @darcywong00 commented on this pull request.
      
      In release/sil/sil_tai_dam_lao/source/sil_tai_dam_lao.kvks:
      > @@ -0,0 +1,8 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<visualkeyboard>
+  <header>
+    <version>10.0</version>
+    <kbdname>sil_tai_dam_lao</kbdname>
+    <flags/>

      This OSK looks blank. When I tried to "Fill from layout" in
        Keyman Developer, it seemed to hang on Importing
          Unshifted K_SLASH.
    
    Fixed.
    
    To avoid this problem in the future, if someone renames a kvks
      file, it may be advisable to first close the project in Developer,
      and then update the name of the name of the kvks by editing
      the kmn file with a text editor. I had renamed the source
    file (i.e. git mv tai_dam_lao.kvks sil_tai_dam_lao.kvks). It appears
    that at that time the kvks file got stripped down to only the file
    header. As near as I can tell, this must have happened because there
    was a few minutes delay between the file rename, and my editing the
    file name in the store(&visualkeyboard) entry in the kmn file.
    During that gap, Keyman Developer could not find the kvks file.
